scripts/content.js

```diff
@@ -1,125 +1,128 @@
-    browser.storage.local.get('35pokes-month').then(monthItems => {
-        browser.storage.local.get('35pokes-year').then(yearItems => {
-            browser.storage.local.get('35pokes-text').then(textItems => {
-                const xhr = new XMLHttpRequest();
-                let fileName;
-                if (monthItems['35pokes-month'] && yearItems['35pokes-year']) {
-                    textSuffix = textItems['35pokes-text'] ? '_' + textItems['35pokes-text'] : '';
-                    fileName = 'https://samuel-peter-chowdhury.github.io/35PokesShowdownFilter/dates/' + yearItems['35pokes-year'] + '_' + monthItems['35pokes-month'] + textSuffix + '.json';
-                } else {
-                    fileName = 'https://samuel-peter-chowdhury.github.io/35PokesShowdownFilter/dates/2024_5.json';
-                }
-                console.log(fileName);
-                xhr.open('GET', fileName, true);
-                xhr.onreadystatechange = function() {
-                    if (xhr.readyState == 4) {
-                        if (xhr.status === 200) {
-                            const data = JSON.parse(xhr.responseText);
-                            data.forEach(item => {
-                                allowedMap.set(item.toLowerCase(), true);
-                            });
-                            browser.storage.local.set({ '35pokes-code': getChallengeCode() }, function() {});
-                            console.log(allowedMap);
-                        } else {
-                            alert('Invalid Date');
-                        }
-                    }
-                }
-                xhr.send();
-            });
-        });
-    });
-}
-
-// Call the function to fetch JSON data
-fetchAllowedPokemonData();
+if (typeof browser === "undefined") globalThis.browser = chrome;
 
+let metaMonth, metaYear, metaText;
+let allowedMap = new Map();
 let removedElements = [];
-
 const observer = new MutationObserver(onMutation);
-browser.storage.local.get(['35pokes-toggleState']).then(items => {
-    if (items['35pokes-toggleState']) {
-        observer.observe(document, {
+
+// Receive settings as the user changes them.
+browser.runtime.onMessage.addListener((message) => {
+    // Toggle state changed
+    if(typeof(message.toggleSet) === "boolean") {
+        if(message.toggleSet) observer.observe(document, {
             childList: true,
             subtree: true,
         });
+        else observer.disconnect();
+        return;
     }
+
+    // Selected meta changed
+    allowedMap.clear();
+    if(message.list) message.list.forEach(item => allowedMap.set(item.toLowerCase()));
+    metaMonth = message.month;
+    metaYear = message.year;
+    metaText = message.text;
+});
+
+// Receive settings on load from last time if present, otherwise load defaults.
+browser.storage.local.get({
+    '35pokes-toggleState': false,
+    '35pokes-list': [],
+    '35pokes-month': 5,
+    '35pokes-year': 2024,

```

Comment:
Use constant for default state

Reply:
This part looks fine, please elaborate if not already resolved?